inference-engine/src/mkldnn_plugin/nodes/batch_to_space.cpp

```diff
@@ -3,219 +3,247 @@
-                    out_dims,
-                    layer->outData[0]->getTensorDesc().getLayout());
-            config.outConfs.push_back(outConfig);
-            config.dynBatchSupport = false;
-            confs.push_back(config);
+                DataConfig outConfig;
+                outConfig.desc = TensorDesc(precision, outDims, layout);
+                config.outConfs.push_back(outConfig);
+
+                config.dynBatchSupport = false;
+                confs.push_back(config);
+            };
+
+            createConfig(inDims.size() == 4 ? NHWC : NDHWC);
+            createConfig(TensorDesc::getLayoutByDims(inDims));
+
+            std::vector<std::pair<ConfLayout, ConfLayout>>  blockConfs { };
+            if (inDims[1] % 8 == 0)  blockConfs.push_back({ConfLayout::BLK8, ConfLayout::BLK8});
+            if (inDims[1] % 16 == 0) blockConfs.push_back({ConfLayout::BLK16, ConfLayout::BLK16});
+            for (auto conf : blockConfs) {
+                addConfig(layer, {DataConfigurator(conf.first, precision),
+                                  DataConfigurator(ConfLayout::ANY, batchToSpaceLayer->insData[1].lock()->getPrecision()),
+                                  DataConfigurator(ConfLayout::ANY, batchToSpaceLayer->insData[2].lock()->getPrecision()),
+                                  DataConfigurator(ConfLayout::ANY, batchToSpaceLayer->insData[3].lock()->getPrecision())},
+                          {DataConfigurator(conf.second, precision)});
+            }
         } catch (InferenceEngine::Exception &ex) {
             errorMsg = ex.what();
         }
     }
-
-    StatusCode execute(std::vector<Blob::Ptr>& inputs, std::vector<Blob::Ptr>& outputs, ResponseDesc *resp) noexcept override {
+    StatusCode execute(std::vector<Blob::Ptr> &inputs, std::vector<Blob::Ptr> &outputs, ResponseDesc *resp) noexcept override {
         switch (inputs[0]->getTensorDesc().getPrecision().size()) {
-            case 1: {
-                process_data<PrecisionTrait<Precision::U8>::value_type>(inputs, outputs);
-                break;
-            }
-            case 2: {
-                process_data<PrecisionTrait<Precision::U16>::value_type>(inputs, outputs);
-                break;
-            }
-            case 4: {
-                process_data<PrecisionTrait<Precision::I32>::value_type>(inputs, outputs);
-                break;
-            }
-            case 8: {
-                process_data<PrecisionTrait<Precision::U64>::value_type>(inputs, outputs);
-                break;
-            }
+            case 1: batchToSpaceKernel<PrecisionTrait<Precision::U8>::value_type> (inputs, outputs);  break;
+            case 2: batchToSpaceKernel<PrecisionTrait<Precision::U16>::value_type>(inputs, outputs); break;
+            case 4: batchToSpaceKernel<PrecisionTrait<Precision::I32>::value_type>(inputs, outputs); break;
+            case 8: batchToSpaceKernel<PrecisionTrait<Precision::U64>::value_type>(inputs, outputs); break;
```

Comment:
The plug-in doesn't support 8 bytes data types in general for now, so the instance should be removed to decrease binary size.